### DISCUSSION TITLE
`borg-remove` doesn't remove the section in .gitmodules

### DISCUSSION THREAD

**Discussion by dalugm** [emacscollective/borg#143] [Support] [answered]
After using `borg-assimilate` to assimilate a new package, using `borg-remove` deletes the clone in working tree but leaves the package's section in `.gitmodules`.

Using `magit-submodule-remove` will delete the section inside `.gitmodules`.

Any ideas?

---

Emmm, seems like `borg-drones` returns a list of atoms with a prefix `/`, so this line doesn't match the corresponding drone:

https://github.com/emacscollective/borg/blob/f9b34133df7ccb650f7e34f43a996c2467def8a3/borg.el#L1085

I don't if this problem is just met by me, here is my env:

```
$ git --version
git version 2.41.0

$ emacs --version
GNU Emacs 30.0.50
Development version 207cd4c23570 on master branch; build date 2023-07-17.
Copyright (C) 2023 Free Software Foundation, Inc.
GNU Emacs comes with ABSOLUTELY NO WARRANTY.
You may redistribute copies of GNU Emacs
under the terms of the GNU General Public License.
For more information about these matters, see the file named COPYING.
```

In Emacs:

``` emacs-lisp
(borg-drones)
;; => ("/ace-window" "/all-the-icons" "/apheleia" "/async"...

;; And (borg-read-clone) returns an atom without a prefix `/'.
```

**Comment by dalugm** [comment]
Well, I guess problem is here:

https://github.com/emacscollective/borg/blob/f9b34133df7ccb650f7e34f43a996c2467def8a3/borg.el#L314

The value of `prefix` in my config is `lib`

https://github.com/emacscollective/borg/blob/f9b34133df7ccb650f7e34f43a996c2467def8a3/borg.el#L350

And here the `offset` (length of prefix) is 3, the `file` is `lib/<package-name>`, so the return list's atom has a prefix `/`.

If I changed it to this, everything works fine:

```diff
- (list (substring file offset)))))
+ (list (substring file (1+ offset))))))
```

**Comment by dalugm** [comment, marked as answer]
Oh, I figure it out that's my config's problem.

I set `borg-drones-directory` to a string without a `/` suffix (I used `expand-file-name` without `file-name-directory`, so the `prefix` is `lib` instead of `lib/`)

**Reply by tarsius** [reply]
Never-the-less I made it a bit safer with e29d9839390c0f9e6c375acb4374430cf5595fdf.
